src/main/java/org/folio/rest/impl/OaiPmhImpl.java

```diff
@@ -87,7 +92,19 @@ public void getOaiIdentifiers(String resumptionToken, String from, String until,
   @Override
   public void getOaiMetadataFormats(String identifier, Map<String, String> okapiHeaders,
                                     Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
-    asyncResultHandler.handle(succeededFuture(GetOaiMetadataFormatsResponse.respond500WithTextPlain("The verb ListMetadataFormats is not supported yet :(")));
+
+    Request request = Request.builder().identifier(identifier).build();
+    HttpClientInterface client = getHttpClient(okapiHeaders);
+    GetOaiMetadataFormatsHelper helper = new GetOaiMetadataFormatsHelper(client, vertxContext, okapiHeaders);
+
+    helper.handle(request, vertxContext)
+      .thenAccept(oaipmh -> {
+        logger.info("Successfully retrieved ListMetadataFormats info: " + oaipmh);
+        asyncResultHandler.handle(succeededFuture(GetOaiRepositoryInfoResponse.respond200WithApplicationXml(oaipmh)));
```

Comment:
I think GetOaiMetadataFormatsResponse should be used here.

Reply:
Fixed. Thanks!